content/blog/what-is-an-algorithm/index.md

```diff
@@ -0,0 +1,212 @@
+
+The same thing goes for a computer program, sometimes, just because there are a few lines of code, doesn’t mean it’s gonna run the fastest or be the most performant.
+
+But before we can get into what makes an algorithm fast or better suited for one task or the other, let's first look at what an algorithm can be made of.
+
+# Building blocks of an algorithm
+
+Let’s take a look at an example that has something more to do with computers: *a basic arithmetic operation.*
+
+If you were asked to calculate 1 + 2, it would take you less than a second, because in your brain “it’s obviously 3”, right? But what if you were asked to break down that operation into steps like we did in the previous example?
+
+If we follow the same thought process as before, we could write something like this:
+
+```elixir
+START
+	- Take the number 1
+	- Take the number 2
+	- Add them together
+	- Obtain the number 3
+END
+```
+
+That's a pretty simple algorithm. But for a computer to be able to take two values and apply an arithmetic operation between them, it needs to store them somewhere first before it does anything with them, which brings us to out next point: "**Variables**"
+
+## Variables
+
+In any computer program, we often have to temporarily store values for different operations. These values can be inputs _(from the user through the keyboard)_ or from the computer storage, or sometimes even values from other operations made by the computer itself as shown in this example:
+
+**_Calculating 2 to the power of 3_**:
+```elixir
+START
+	- Take the number 2
+	- Take another number 2
+	- Multiply 2 by 2
+	- Obtain the number 4
+	- Take another number 2
+	- Multiply 4 by 2
+	- Obtain the number 8
+END
+```
+
+In the fourth step, we have to store the value **4** which is returned by the computer itself, in order to multiply it again by the number **2** to get the final result.
+
+> Values stored by a computer don’t always have to be numbers though, they can also be letters *(known in programming languages as characters, like a, b, k, y…)*, words *(known as “strings”, which means “strings of characters”)*, and other types of variables that you’ll meet later on as you deal with programming languages.
+
+Whenever the computer needs to store a value of any kind, it uses what we call a “Variable”.
+
+A variable can be initially considered as a box to keep things in. And since there can be millions of "boxes" inside a computer, we need to differentiate them, which is why each one has a unique name, address (where it's kept in the memory), etc.
+
+So, for any value that needs to be used, we can ask the computer to create a unique **variable** and **assign** our value to that variable so that we're able to use it throughout the code.
```

Comment:
Can we see an example of one of these "boxes"?

AKA:

```elixir
START
	- Take the number 2
	- Take another number 2
	- Multiply 2 by 2
	- Obtain the number 4, and put it in a "box" called "result"
	- Take another number 2
	- Multiply the value inside of the box "result" (4) by 2
	- Obtain the number 8
END
```

Reply:
Added another illustration along with re-writing the pseudo-code to go along with it